platform_login/test/test.js

```diff
@@ -15,13 +15,102 @@
 
 const assert = require('chai').assert;
+const expect = require('chai').expect;
+const awsContext = require('aws-lambda-mock-context');
+const AWSCognito = require('amazon-cognito-identity-js');
+const sinon = require('sinon');
 const index = require('../index');
 
-describe('Sample', function() {
-  it('tests handler', function(done) {
+var event, context, stub, spy, callback;
+
+//Setting up a spy to wrap mocked cognito functions (stubs) for each test scenario
+spy = sinon.spy();
+
+describe('Login handler', function() {
+
+  //Setting up default values for the aws event and context needed for handler params
+  beforeEach(function(){
+    event = { "method" : "POST",
+              "stage" : "test",
+              "body" : { "username" : "whatTimeIsIt",
+                         "password" : "AdventureT1me!"
+                       }
+            };
+    context = awsContext();
+    callback = (value) => {
+      return value;
+    };
+  });
+
+  /*
+  * Given an event with no method, handler() shows that a Bad Request has been made
+  * @param {object} event containing only stage and body attributes
+  * @param {object} aws context
+  * @param {function} callback function that returns what was passed
+  * @returns {string} callback function showing error type of badrequest has occured
+  */
+  it("should throw a BadRequest error for undefined method", function(){
+    event.method = undefined;
+    var bool = index.handler(event,context,callback).includes("Bad Request");
+    assert.isTrue(bool);
+  });
+
+  /*
+  * Given an event with non POST method, handler() shows that a Bad Request has been made
+  * @param {object} event containing a non POST method
+  * @param {object} aws context
+  * @param {function} callback function that returns what was passed
+  * @returns {string} callback function showing error type of badrequest has occured
+  */
+  it("should throw a BadRequest error for method other than POST", function(){
+    var methods = ["PUT","GET","HEAD","OPTIONS","DELETE"];
+    var badRequestBool = true;
+    //check if BadRequest occurs for each http method, if one passes without a bad request error
+    //have the above bool value be false
+    for(var i=0; i < methods.length; i++){
+      event.method = methods[i];
+      if(!index.handler(event,context,callback).includes("Bad Request")){
+        badRequestBool = false;
+      }
+    }
+    assert.isTrue(badRequestBool);
+  });
+
+  /*
+  * Given an event with no username, handler() shows that a 101 error has been made
+  * @param {object} event without username information
+  * @param {object} aws context
+  * @param {function} callback function that returns what was passed
+  * @returns {string} callback function showing username was not provided
+  */
+  it("should throw a 101 error for missing username", function(){
+    event.body.username = undefined;
+    var bool = index.handler(event,context,callback).includes("101");

```

Comment:
We also need to assert that we have a non-empty error message for all the failure scenarios as this is used by UI to display errors.